src/pages/ProfilesExplorerView/components/SceneExploreDiffFlameGraph/components/SceneComparePanel/SceneComparePanel.tsx

```diff
@@ -211,16 +232,65 @@ export class SceneComparePanel extends SceneObjectBase<SceneComparePanelState> {
     return (this.state.timeseriesPanel.state.body.state.$timeRange as SceneTimeRangeWithAnnotations).useState();
   }
 
+  applyPreset({ from, to, diffFrom, diffTo, label }: Preset) {
+    this.setDiffRange(diffFrom, diffTo);
+
+    this.state.$timeRange.setState(buildTimeRange(from, to));
+
+    this.updateTitle(label);
+  }
+
+  setDiffRange(diffFrom: string, diffTo: string) {
+    const $diffTimeRange = this.state.timeseriesPanel.state.body.state.$timeRange as SceneTimeRangeWithAnnotations;
```

Comment:
**thought:** Should pull the state up?

Drilling down the state looks a bit cumbersome and makes testing harder do to mocking. Do you think it would make sense / be feasible to pull `timeRange` state up and save it only in SceneComparePanel. Timeseries panel looks for nearest timeRange anyway so will use the same one. This is not needed in this PR, just food for thought.

Reply:
As we discussed face to face: this time range is actually a timerange-like object (a hack) that we use to enable annotations on the time series (see [SceneTimeRangeWithAnnotations](https://github.com/grafana/explore-profiles/blob/main/src/pages/ProfilesExplorerView/components/SceneExploreDiffFlameGraph/components/SceneComparePanel/components/SceneTimeRangeWithAnnotations.ts)). So it makes sense that we keep it as close as possible as where it's used (it's also responsible to change the data received from the API on-the-fly, in order to add the annotation on the time series). 

Also, if in the future the platform provides a way to add range selections out-of-the-box (via a new API or so), it's likely that it's going to be configured on the timeseries panel.

Finally, for testing, I wouldn't mock anything ; I'd use end-to-end testing as we do now because these user interactions are quite complex (because they involve many different components).